dlp/snippets/deid.py

```diff
@@ -1023,6 +1023,122 @@ def deidentify_with_exception_list(
+
+    Returns:
+        De-identified table is returned;
+        the response from the API is also printed to the terminal.
+
+    Example:
+    table_data = {
+        "header":[
+            "name"
+            "email",
+            "phone number"
+        ],
+        "rows":[
+            [
+                "Robert Frost",
+                "robertfrost@xyz.com",
+                "4232342345"
+            ],
+            [
+                "John Doe",
+                "johndoe@pqr.com",
+                "4253458383"
+            ]
+        ]
+    }
+
+    >> $ python deid.py table_replace_with_infotype \
+        '{"header": ["name", "email", "phone number"],
+        "rows": [["Robert Frost", "robertfrost@xyz.com", "4232342345"],
+        ["John Doe", "johndoe@pqr.com", "4253458383"]]}' \
+        ["PERSON_NAME"] ["name"]
+        >> '{"header": ["name", "email", "phone number"],
+            "rows": [["[PERSON_NAME]", "robertfrost@xyz.com", "4232342345"],
+            ["[PERSON_NAME]", "johndoe@pqr.com", "4253458383"]]}'
+    """
+
+    # Import the client library
+    import google.cloud.dlp
+
+    # Instantiate a client.
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Construct the `table`. For more details on the table schema, please see
+    # https://cloud.google.com/dlp/docs/reference/rest/v2/ContentItem#Table
+    headers = [{"name": val} for val in table_data["header"]]
+    rows = []
+    for row in table_data["rows"]:
+        rows.append({"values": [{"string_value": cell_val} for cell_val in row]})
+
+    table = {"headers": headers, "rows": rows}
+
+    # Construct item
+    item = {"table": table}
+
+    # Specify fields to be de-identified
+    deid_content_list = [{"name": _i} for _i in deid_content_list]
+
+    # Construct inspect configuration dictionary
+    inspect_config = {"info_types": [{"name": info_type} for info_type in info_types]}
+
+    # Construct deidentify configuration dictionary
+    deidentify_config = {
+        "record_transformations": {
+            "field_transformations": [
+                {
+                    "info_type_transformations": {
+                        "transformations": [
+                            {
+                                "primitive_transformation": {"replace_with_info_type_config": {}}
+                            }
+                        ]
+                    },
+                    "fields": deid_content_list,
+                }
+            ]
+        }
+    }
+
+    # Convert the project id into a full resource id.
+    parent = f"projects/{project}"
+
+    # Call the API.
+    response = dlp.deidentify_content(
+        request={
+            "parent": parent,
+            "deidentify_config": deidentify_config,
+            "item": item,
+            "inspect_config": inspect_config
+        })
+
+    # Print the result
+    print("Table after de-identification: {}".format(response.item.table))
```

Comment:
please switch this to f-strings

Reply:
thank you for reviewing, Updated the code as per your comment, please review. Thank you